src/services/healthcareProfessionalService.ts

```diff
@@ -0,0 +1,1033 @@
+            error: {
+                field: 'facilityIds',
+                httpStatus: 400
+            } 
+        }
+    }
+    if (creations.length === 0 && deletions.length > 0) {
+        return { newFacilityId: null, error: { field: 'facilityIds', httpStatus: 400 } }
+    }
+    if (creations.length === 1) { 
+        return {
+            newFacilityId: creations[0].otherEntityId
+        }
+    }
+    if (deletions.length > 0) {
+        return {
+            newFacilityId: null
+        }
+    }
+    return {
+        newFacilityId: null
+    }
+}
+
+/**
+ * Gets the Healthcare Professional from the database that matches on the id.
+ * @param id A string that matches the id of the Supabase for the professional.
+ * @returns A Healthcare Professional object.
+ */
+export async function getHealthcareProfessionalById(
+    id: string
+): Promise<Result<gqlTypes.HealthcareProfessional>> {
+    try {
+        // Validate the incoming id
+        const validationResult = validateIdInput(id)
+
+        if (validationResult.hasErrors) {
+            logger.warn(`Validation Error: User passed invalid HP id: ${id}`)
+
+            return {
+                data: {} as gqlTypes.HealthcareProfessional,
+                hasErrors: true,
+                errors: (validationResult.errors ?? []).map(err => ({
+                    ...err,
+                    field: 'getHealthcareProfessionalById'
+                }))
+            }
+        }
+
+        const supabase = getSupabaseClient()
+
+        // Fetch the HP row by primary key; .single() requires exactly one row
+        const { data: hpRow, error: hpRowError } = await supabase
+            .from('hps')
+            .select('*')
+            .eq('id', id)
+            .single()
+
+        // Supabase returns PGRST116 when .single() finds no rows.
+        if (hpRowError?.code === 'PGRST116' || !hpRow) {
+            return {
+                data: {} as gqlTypes.HealthcareProfessional,
+                hasErrors: true,
+                errors: [{
+                    field: 'getHealthcareProfessionalById',
+                    errorCode: ErrorCode.INTERNAL_SERVER_ERROR,
+                    httpStatus: 500
+                }]
+            }
+        }
+
+        if (hpRowError) {
+            throw hpRowError
+        }
+
+        // Load relations from junction table (facility links for this HP)
+        const { data: facilityLinkRows, error: facilityLinkRowsError } = await supabase
+            .from('hps_facilities')
+            .select('facilities_id')
+            .eq('hps_id', id)
+
+        if (facilityLinkRowsError) {
+            throw facilityLinkRowsError
+        }
+
+        // Normalize facility IDs to a string array
+        const facilityIds = (facilityLinkRows ?? []).map(row => row.facilities_id as string)
+
+        // Cast DB row to internal schema and map to GraphQL shape including relations
+        const dbHp = hpRow as dbSchema.DbHealthcareProfessionalRow
```

Comment:
we shouldn't call them rows since we're returning back objects or domain entities. Take a look at how we previously named them in the old files if that helps